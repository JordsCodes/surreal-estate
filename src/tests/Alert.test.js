import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("displays error message", () => {
    const component = render(<Alert message="Error!" />);
    const alert = component.getByText("Error!");

    expect(alert.textContent).toBe("Error!");
    expect(alert).toMatchSnapshot();
  });
  it("displays success message", () => {
    const component = render(<Alert message="Success!!!" success />);
    const alert = component.getByText("Success!!!");

    expect(alert.textContent).toBe("Success!!!");
    expect(alert).toMatchSnapshot();
  });
  it("doesn't render if message is null", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();

    expect(alert).toMatchSnapshot();
  });
});
