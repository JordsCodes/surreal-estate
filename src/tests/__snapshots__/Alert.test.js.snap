// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays error message 1`] = `
<div
  class="alert-error"
>
  Error!
</div>
`;

exports[`Alert displays success message 1`] = `
<div
  class="alert-success"
>
  Success!!!
</div>
`;

exports[`Alert doesn't render if message is null 1`] = `<DocumentFragment />`;
